app/controllers/candidate_interface/gcse_details_controller.rb

```diff
@@ -1,15 +1,83 @@
 module CandidateInterface
   class GcseDetailsController < CandidateInterfaceController
-    def edit
-      subject = params[:subject]
+    # 1th step - Edit qualification type
+    def edit_type
+      @subject = subject_param
```

Comment:
Could we create a `before_action` for setting `@subject` as it's used in all methods?

Reply:
I definitely agree with you, `before_action` should be used with caution! But in this case I think it's the perfect use of it.

A nitpick, but I tend to see `before_action` used like this `before_action :set_subject` and then a private method defined.